cluster-state-service/internal/features/wrappers/ec2_wrapper.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2016-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package wrappers
+
+import (
+	"encoding/base64"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/pkg/errors"
+)
+
+type EC2Wrapper struct {
+	client *ec2.EC2
+}
+
+func NewEC2Wrapper() EC2Wrapper {
+	awsSession := newAWSEC2Session()
+	return EC2Wrapper{
+		client: ec2.New(awsSession),
+	}
+}
+
+func newAWSEC2Session() *session.Session {
+	var sess *session.Session
+	var err error
+	if endpoint, err := getEC2Endpoint(); err != nil {
+		sess, err = session.NewSession()
+	} else {
+		sess, err = session.NewSessionWithOptions(session.Options{
+			Config: aws.Config{
+				Endpoint: aws.String(endpoint),
+			},
+		})
+	}
+	if err != nil {
+		panic(err)
+	}
+	return sess
+}
+
+func (ec2Wrapper EC2Wrapper) RunInstance(count *int64, clusterName *string, instanceProfileName *string, keyName *string) ([]*string, error) {
+
+	currentDate := `$(date "+%s")`
+	utmpStartDate := `$(last -t 19701111111111 | cut -d" " -f3-)`
+	bootTime := `$(date -d "` + utmpStartDate + `" "+%s")`
+	userData := `#!/bin/bash
+echo ECS_CLUSTER=` + *clusterName + ` >> /etc/ecs/ecs.config
+#cloud-boothook
+#!/bin/sh -x
+echo -e '#!/bin/bash' >> /etc/cron.hourly/shutdowninstance
+echo -e '[ "$(( ` + currentDate + ` - ` + bootTime + ` ))" -gt 3600 ] && /sbin/shutdown -h now' >> /etc/cron.hourly/shutdowninstance
+chmod +x /etc/cron.hourly/shutdowninstance`
+	encodedUserData := base64.StdEncoding.EncodeToString([]byte(userData))
+
+	amiID := "ami-022b9262"
```

Comment:
The following URL lists the most up-to-date ECS AMIs by region:
http://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-optimized_AMI.html